### PR TITLE
grape-jdk support mac

### PR DESCRIPTION
**Pull request by zhanglei1949**
## What do these changes do?

Add GRAPE-jdk support for mac os. Including m1 chipset.

## Related issue number

#1763 

Fixes

### CODE REVIEW

**Comment by zhanglei1949** [comment]
For reviews:
- ENABLE_JAVA_SDK turned on by default.
- Detect mac Darwin in CMakeLists.
- Turn off ld.lld off on mac, since not supported
- Install openjdk@11 from brew on mac Darwin.
- Fixed hard-coded jni library loading.

**Comment by zhanglei1949** [comment]
> 

Please evaluate the necessity of turning enable_java_sdk=on by default, thanks.

File: analytical_engine/CMakeLists.txt [hunk 1]
```diff
@@ -11,21 +11,22 @@ project(analytical_engine LANGUAGES C CXX VERSION ${GRAPHSCOPE_ANALYTICAL_VERSIO
...
+option(ENABLE_JAVA_SDK "Build with support for java sdk" ON)
 
 # Get system processor
 execute_process(COMMAND uname -m OUTPUT_VARIABLE SYSTEM_PROCESSOR)
 string(REGEX REPLACE "\n$" "" SYSTEM_PROCESSOR "${SYSTEM_PROCESSOR}")
 
-if (APPLE AND ENABLE_JAVA_SDK)
-    message(FATAL_ERROR "Java SDK is currently not supported on macos.")
+if (CMAKE_HOST_SYSTEM_NAME MATCHES "Darwin")
+    set(SYSTEM_PROCESSOR "arm64")
```
Comment (lidongze0629) [inline analytical_engine/CMakeLists.txt:21]: macOS with Intel chip also need to be supported.

Comment (zhanglei1949) [reply analytical_engine/CMakeLists.txt:21]: macOS with x86 chip should be supported. Maybe I should delete this check, since it not necessary.

**Review by lidongze0629 — APPROVED** [review]

**Pull request merged**